### PR TITLE
Add mqtt network buffer size check in OTA demo

### PR DESCRIPTION
**Pull request by ravibhagavandas**
Added a compile time check to verify mqtt network buffer size is enough to store largest OTA block.
Removed unused includes.

### CODE REVIEW

File: source/ota_over_mqtt_demo.c [hunk 1]
```diff
@@ -80,6 +76,10 @@
     #error "Please define the democonfigCLIENT_IDENTIFIER with the thing name for which OTA is performed"
 #endif
 
+#if !defined( MQTT_AGENT_NETWORK_BUFFER_SIZE ) || ( MQTT_AGENT_NETWORK_BUFFER_SIZE < OTA_DATA_BLOCK_SIZE )
```
Comment (pvyawaha) [inline source/ota_over_mqtt_demo.c:79]: As this is OTA demo why not just set the buffer size from the ota data block size instead of throwing error?

**Review by pvyawaha — APPROVED** [review]

File: source/ota_over_mqtt_demo.c (hunk 1 above)
Comment (ravibhagavandas) [reply source/ota_over_mqtt_demo.c:79]: MQTT_AGENT_NETWORK_BUFFER_SIZE is used by all demos sharing the MQTT agent connection. Buffer size is set irrespective of OTA demo task.

File: source/ota_over_mqtt_demo.c [hunk 2]
```diff
@@ -80,6 +76,10 @@
...
+    #error "Please define MQTT_AGENT_NETWORK_BUFFER_SIZE to adequate size as required by OTA data block"
```
Comment (dan4thewin) [inline source/ota_over_mqtt_demo.c:80]: Could we reword this to be a little clearer?  e.g., buffer is too small, please increase the size to at least x...

Comment (ravibhagavandas) [reply source/ota_over_mqtt_demo.c]: Reworded to be more clear.

**Review by dan4thewin — APPROVED** [review]

**Review by pvyawaha — APPROVED** [review]
